### PR TITLE
Move student update

### PR DESCRIPTION
**Pull request by emcelroy**
- A new Tool model is added, this replaces the direct source_type on external_activities. 
- External Reports now have two new fields for supporting moving students. These are added to the external report instead of the Tool so the same report service could be used for different tools. And so one tool could have multiple reports. In the future it might be necessary to tell the tool about the move as well. Here are two new fields:
  - move_students_api_token
  - move_students_api_url
- When a student is moved the portal send the following JSON structure to all external reports that have a move_students_api_url
```
{
  new_class_info_url: [portal_url to class info of new class],
  new_context_id: [portal class_hash for the new class],
  old_context_id: [portal class_hash for the old class],
  platform_id: [portal platform_id which is its site_url[,
  platform_user_id: [portal user id]
  assignments: [
    {
      new_resource_link_id: [new portal offering id],
      old_resource_link_id: [old portal offering id],
      tool_id: [url to lara (or whatever tool is specified for the resource)]
    },
    ...
  ]   
}
```

- This PR also improves the admin page of the portal.

### CODE REVIEW

File: app/assets/stylesheets/web/admin.scss [hunk 1]
```diff
@@ -137,3 +137,17 @@
...
   }
 }
+
+.admin-links-cols {
+  column-count: 2;
+  margin-bottom: 30px;
+
+  .col {
+    break-inside: avoid-column;
+  }
```
Comment (scytacki) [inline app/assets/stylesheets/web/admin.scss]: This `.col` class looks like it isn't used. Perhaps it should replace the inline style on the project admin div?

Comment (emcelroy) [reply app/assets/stylesheets/web/admin.scss]: Yeah, you're right. Thanks for catching that. I cleaned that up in the latest commit.

**Comment by scytacki** [comment]
Some notes that should be documented in a better place:
- A new Tool model is added, this replaces the direct addition of the source_type to external_activities. 
- External Reports now supports two new fields for supporting moving students. These are added to the external report so the same report service could be used for different tools. Currently it is only the report service that needs to know about moving students. 
  - move_students_api_token
  - move_students_api_url
- When a student is moved all external reports that have a move_students_api_url are sent a JSON structure describing the move.
```
{
  new_class_info_url: [portal_url to class info of new class],
  new_context_id: [portal class_hash for the new class],
  old_context_id: [portal class_hash for the old class],
  platform_id: [portal platform_id which is its site_url[,
  platform_user_id: [portal user id]
  assignments: [
    {
      new_resource_link_id: [new portal offering id],
      old_resource_link_id: [old portal offering id],
      tool_id: [url to lara (or whatever tool is specified for the resource)]
    },
    ...
  ]   
}
```

- This PR also improves the admin page of the portal.

File: app/models/portal/offering.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Portal::Offering < ActiveRecord::Base
 
   has_many :activity_feedbacks, class_name: "Portal::OfferingActivityFeedback", foreign_key: "portal_offering_id"
 
-  [:name, :short_description, :long_description, :long_description_for_teacher, :icon_image].each { |m| delegate m, :to => :runnable }
+  [:name, :short_description, :long_description, :long_description_for_teacher, :icon_image, :tool_id].each { |m| delegate m, :to => :runnable }
```
[the remark this answers is not in the document]
Comment (emcelroy) [reply app/models/portal/offering.rb]: Is the change in the latest commit all that needed to be done for this?

[the remark this answers is not in the document]
Comment (scytacki) [reply app/models/portal/offering.rb]: Yup that is it.

**Pull request merged**